### DISCUSSION TITLE
Run renovate in dry-run against specific branch?

### DISCUSSION THREAD

**Discussion by HenrikPoulsen** [renovatebot/renovate#16108] [Request Help] [answered]
Is there a way to configure renovate to use the configuration from a specific branch instead of always using the mainline one?

In my case we run renovate as a github workflow in each repo. We have a "Renovate Validation" github workflow which people can run in their branches, which runs the config validation on it.
We would also like to be able to run a dry-run on that branch when the renovate.json has changed to verify that the changes will work when it has landed.

It's happening surprisingly often that people request this functionality internally and it doesn't look like it's supported.
I could imagine that one perhaps could use some environment variable or such to pass to renovate to read the config from the current branch the workflow is run on (strictly in dry-run mode). But I can imagine this is not trivial to do.

The workflows people have today is that either they blindly merge the changes and hope everything is fine. Or they fork the repo and make changes on the mainline branch of the fork to test it by running renovate manually locally etc.
It's not ideal.

Do you have any ideas or suggestions here? Would it make sense at all to be able to override the branch the config reads from?

**Comment by rarkins** [comment, marked as answer]
Renovate always looks in the default branch for config initially. You can't override that. But you can use `baseBranches` plus https://docs.renovatebot.com/configuration-options/#usebasebranchconfig to potentially achieve what you want, which means you can specify that whatever config is in that base branch overrides what's in the default branch. But an important caveat: the default branch config isn't *ignored* - it's just merged first, allowing the other base branch's config to merge second and override.

**Reply by TWiStErRob** [reply]
Why isn't an option for a full "override"? There's no light-weight way of testing Renovate right now. We have to either create a fork or test in production. I'm running Renovate locally in "self-hosted" mode, but even then I can't run against a specific branch in a specific repo. Note that this means that I cannot remove/replace `packageRules` because they're both applied. Also if I'm the owner of a repo, I can't fork, I need a separate GitHub account just for testing Renovate?! 😨

**Reply by pfragoso** [reply]
Also bumped into this while trying to test renovate locally.  My use-case was simply to try and test changes into the renovate.json file while creating a new branch (ie ":disableDependencyDashboard" + baseBranches ) which seems it's not possible currently.

**Reply by frickerj** [reply]
+1, this would be a very useful feature, surprised it is not already possible

**Reply by joosehav** [reply]
Being able to specify a branch which ignores configurations in base branch would be valuable for validating changes to configurations via dry-run. Suggested solution is better than nothing, but some changes done in a branch might not play nicely with base branch configurations.
